### PR TITLE
add a more precise error message so errors that appear in logs can be…

### PR DESCRIPTION
**Pull request by mikerott**
… confidently triaged

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -4,6 +4,9 @@ services:
     build:
       context: .
       dockerfile: Dockerfile
+    depends_on: 
+      - gateway-inbound
```
Comment (bkrebsbach) [inline docker-compose.yaml]: Task: I'm a big fan on the `depends_on` addition, but I think the `gateway-inbound` container actually depends on the `app` container.

File: pkg/assetfetcher/asset.go [hunk 2]
```diff
@@ -330,7 +333,7 @@ func (a Asset) AssetPayloadToAssetEvent() (domain.AssetEvent, error) {
 type assetHistoryEvents []AssetHistory
 
 func (a assetHistoryEvents) lastScannedTimestamp() (time.Time, error) {
-	latestTime := time.Time{}
+	latestTime := time.Time{} // empty struct
```
Comment (kconwayatlassian) [inline pkg/assetfetcher/asset.go]: nit: the typical go form of this is `var latestTime time.Time` which conveys "a variable with the zero value of type X"

File: pkg/assetfetcher/errors.go [hunk 3]
```diff
@@ -82,3 +82,14 @@ type MissingRequiredFields struct {
...
+
+// AssetNotScanned represents an error when the asset being checked has never been scanned
+type AssetNotScanned struct {
+	AssetID int64
+	AssetIP string
+}
+
+// Error prints a useful message indicating why an asset scan report will not be produced
+func (e *AssetNotScanned) Error() string {
+	return fmt.Sprintf("This Nexpose asset has never been scanned, so no scan report can be produced. ID: %v, IP: %s", e.AssetID, e.AssetIP)
```
Comment (kconwayatlassian) [inline pkg/assetfetcher/errors.go]: nit: go error messages have a suggested format that includes lower case beginnings for strings: https://github.com/golang/go/wiki/Errors

File: docker-compose.yaml (hunk 1 above)
Comment (LorenBoren) [reply docker-compose.yaml]: Question: What's the purpose of adding this? Are things not starting up correctly?

File: pkg/assetfetcher/asset.go [hunk 4]
```diff
@@ -316,9 +316,12 @@ func (a Asset) AssetPayloadToAssetEvent() (domain.AssetEvent, error) {
 	if err != nil {
 		return domain.AssetEvent{}, err
 	}
-	if a.ID == 0 || a.IP == "" || lastScanned.IsZero() {
+	if a.ID == 0 || a.IP == "" {
 		return domain.AssetEvent{}, &MissingRequiredFields{a.ID, a.IP, lastScanned}
 	}
+	if lastScanned.IsZero() { // this Asset (aka "resource" in Nexpose terminology) has never been scanned
```
Comment (austburn) [inline pkg/assetfetcher/asset.go]: nit: I think we're talking "Site" as opposed to "Asset" here

File: docker-compose.yaml (hunk 1 above)
Comment (mikerott) [reply docker-compose.yaml]: Things started fine, but this is just more precise and safe for consumers of this project.

**Pull request merged**